chain/ethereum/src/runtime/runtime_adapter.rs

```diff
@@ -43,13 +43,17 @@ impl blockchain::RuntimeAdapter<Chain> for RuntimeAdapter {
     fn host_fns(&self, ds: &DataSource) -> Result<Vec<HostFn>, Error> {
         let abis = ds.mapping.abis.clone();
         let call_cache = self.call_cache.cheap_clone();
-        let eth_adapter = self
-            .eth_adapters
-            .cheapest_with(&NodeCapabilities {
-                archive: ds.mapping.requires_archive()?,
-                traces: false,
-            })?
-            .cheap_clone();
+        let eth_adapter = match self.eth_adapters.call_only_adapter() {
+            Err(err) => bail!(err),
+            Ok(None) => self
+                .eth_adapters
+                .cheapest_with(&NodeCapabilities {
+                    archive: ds.mapping.requires_archive()?,
```

Comment:
If we are making eth calls, don't we always need an archive node? I am not sure, but my assumption is that this is always `true`

Reply:
This part was already here so I'm not sure if it;s different for other clients that are not ethereum. In the new code I made that assumption